### PR TITLE
Make InvocationProcessor more robust against exceptions

### PR DESCRIPTION
**Pull request by ebr**
1. if retrieving an item from the queue raises an exception, the `InvocationProcessor` thread crashes, but the API continues running in a non-functional state. This fixes the issue
2. when there are no items in the queue, sleep 1 second before checking again - avoid hammering the queue uncontrollably
3. Ensures the thread isn't crashed if an exception is raised from invoker, and emits the error event accordingly (#3222)

Intentionally using base `Exception`s here because for now we don't know which specific exception to expect. We can refine this in the future.

Fixes (sort of)? #3222

### CODE REVIEW

File: invokeai/app/services/processor.py [hunk 1]
```diff
@@ -124,7 +132,16 @@ def __process(self, stop_event: Event):
...
-                    self.__invoker.invoke(graph_execution_state, invoke_all=True)
+                    try:
+                        self.__invoker.invoke(graph_execution_state, invoke_all=True)
+                    except Exception as e:
+                        logger.error("Error while invoking: %s", e)
+                        self.__invoker.services.events.emit_invocation_error(
+                            graph_execution_state_id=graph_execution_state.id,
+                            node=invocation.dict(),
+                            source_node_id=source_node_id,
+                            error=traceback.format_exc(e)
```
Comment (psychedelicious) [inline invokeai/app/services/processor.py:143]:
```suggestion
                            error=traceback.format_exc()
```
Typo, `format_exc()` does not take the exception as an arg

**Review by psychedelicious — CHANGES_REQUESTED** [review]

**Review by psychedelicious — APPROVED** [review]

**Review by blessedcoolant — APPROVED** [review]

File: invokeai/app/services/processor.py [hunk 2]
```diff
@@ -34,8 +36,14 @@ def __process(self, stop_event: Event):
...
             while not stop_event.is_set():
-                queue_item: InvocationQueueItem = self.__invoker.services.queue.get()
+                try:
+                    queue_item: InvocationQueueItem = self.__invoker.services.queue.get()
+                except Exception as e:
+                    logger.debug("Exception while getting from queue: %s" % e)
+
                 if not queue_item:  # Probably stopping
+                    # do not hammer the queue
+                    time.sleep(1)
```
Comment (Kyle0654) [inline invokeai/app/services/processor.py:46]: 1 second might be long for responsiveness. Maybe a shorter period of time?

**Review by Kyle0654 — APPROVED** [review]

File: invokeai/app/services/processor.py (hunk 2 above)
Comment (ebr) [inline invokeai/app/services/processor.py:46]:
```suggestion
                    time.sleep(0.5)
```

Comment (ebr) [reply invokeai/app/services/processor.py]: I reduced it to 0.5s. There doesn't seem to be a perceptible difference, thanks to the invocation progress bar going into "Preparing" mode immediately after hitting "Invoke".

**Pull request merged**